### PR TITLE
Create snyk.yml

### LINKED ISSUE #2691: Set Up Snyk
We now have approval for funding to use Snyk.

## Acceptance Criteria
- [ ] Work with Gray to get the payment details
- [ ] Set up subscription
- [ ] Integrate it with the project

## 🛑 Blockers
```[tasklist]
### Blocked By
- [ ] #5338
````

### PR DESCRIPTION
**Pull request by brindasasi**
Resolves #2691
Scope of this yml file is focused on frontend dependencies 
This snyk test will test for new dependencies that are being added in PRs 
Tested and works fine.
Todo: 

- [x] - Remove Continue on Error flag to make it fail 
- [x] - Look into upload the failure report into code scanning in github 

How to test npm vulnerability scan: 

1.  checkout the branch
2. install pdfimage package locally that is vulnerable (https://snyk.io/advisor/npm-package/pdf-image)
3. Now you can test github action either locally via https://github.com/nektos/act Or push to the branch and test ( but please make sure to revert the commit ) 
4. Github action for snyk should fail showing the package as vulnerable

### CODE REVIEW

File: .github/workflows/snyk.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      - name: Run Snyk to check configuration files for security issues
+        # Snyk can be used to break the build when it detects security issues.
+        # In this case we want to upload the issues to GitHub Code Scanning
+        continue-on-error: true
+        uses: snyk/actions/node@master
+        env:
+          # In order to use the Snyk Action you will need to have a Snyk API token.
+          # More details in https://github.com/snyk/actions#getting-your-snyk-token
+          # or you can signup for free at https://snyk.io/login
+          SNYK_TOKEN: 227a758a-3ae5-4f31-a3fd-64ed1127f06e
```
Comment (patcon) [inline .github/workflows/snyk.yml:48]: Is this ok to be in the public file? Or should we use secrets?

Comment (brindasasi) [reply .github/workflows/snyk.yml:48]: I guess it should be in secrets, will change it ..thx

Comment (esizer) [reply .github/workflows/snyk.yml]: Was this regenerated when it was moved to secrets?

File: .github/workflows/snyk.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+# For more examples, including how to limit scans to only high-severity issues
+# and fail PR checks, see https://github.com/snyk/actions/
+
+name: Snyk
+
+on:
+  push:
+    branches: [ "main" ]
+    paths:
+      - frontend/**
```
Comment (petertgiles) [inline .github/workflows/snyk.yml:19]: It looks like Snyk can also scan PHP.  Do we want to have it scan /api as well?

File: .github/workflows/snyk.yml (hunk 1 above)
Comment (brindasasi) [reply .github/workflows/snyk.yml]: yes ... it got re-generated and then moved to secrets

**Comment by petertgiles** [comment]
I branched a new branch off of your branch: PR #3645  But it didn't fail?

**Comment by mnigh** [comment]
@brindasasi 
1. should **high priority** be removed from this PR since it's not in one of the active columns (Sprint Backlog, In Progress, Review/QA)?
2. should this PR be marked as a **draft** since it does not seem ready for review and is not even in the Review/QA column?

**Comment by mnigh** [comment]
@brindasasi is this PR still needed or can it be closed? it seems like snyk is being run in our checks currently:

## Screenshot
Screen Shot 2022-11-23 at 12 22 48

**Comment by patcon** [comment]
Yeah, a little confused the difference, but are snyk actions diff than apps?

It seems like this should be showing up in Security tab:
https://snyk.io/blog/snyk-security-information-in-github-code-scanning/

But then there's nothing in there:
https://github.com/GCTC-NTGC/gc-digital-talent/security/code-scanning
Screen Shot 2022-11-30 at 3 54 23 PM

And there's no indication of any third-party snyk apps enabled on this repo:
https://github.com/GCTC-NTGC/gc-digital-talent/settings/installations
Screen Shot 2022-11-30 at 3 53 35 PM

And the UI seems to indicate we can still add snyk apps, but as actions only (as in, it's not enabled in workflows yet):
https://github.com/GCTC-NTGC/gc-digital-talent/actions/new?category=none&query=snyk

So I'm wondering if we're in some limbo state between a migration? Or maybe a user added it to the org, and we can't see it on the repo?
https://docs.snyk.io/integrations/git-repository-scm-integrations/github-integration

It seems like the current way is NOT how we want it set up, since it's enabled in a very phantom way...!

**Comment by brindasasi** [comment]
we have this snyk issue put into icebox. so closing it for now.

**Pull request closed without merging**